### ISSUE TITLE
Build and Compile test suite of Spoon

### ISSUE THREAD

**Issue by danglotb** [INRIA/spoon#1880]
Hi,

I am trying to build and compile the test suite of Spoon, with Spoon.

I am getting the following errors:

```java
spoon.support.SpoonClassNotFoundException: cannot load class: spoon.generating.meta.$Role$
	at spoon.support.reflect.reference.CtTypeReferenceImpl.findClass(CtTypeReferenceImpl.java:138)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.getActualClass(CtTypeReferenceImpl.java:120)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.getModifiers(CtTypeReferenceImpl.java:465)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.canAccess(CtTypeReferenceImpl.java:595)
	at spoon.reflect.visitor.ImportScannerImpl.addClassImport(ImportScannerImpl.java:221)
	at spoon.reflect.visitor.MinimalImportScanner.addClassImport(MinimalImportScanner.java:43)
	at spoon.reflect.visitor.ImportScannerImpl.computeImports(ImportScannerImpl.java:168)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.computeImports(DefaultJavaPrettyPrinter.java:193)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.calculate(DefaultJavaPrettyPrinter.java:1805)
	at spoon.support.compiler.jdt.CompilationUnitWrapper.getContents(CompilationUnitWrapper.java:34)
	at org.eclipse.jdt.internal.compiler.parser.Parser.parse(Parser.java:11898)
	at org.eclipse.jdt.internal.compiler.parser.Parser.parse(Parser.java:11870)
	at org.eclipse.jdt.internal.compiler.parser.Parser.dietParse(Parser.java:10280)
	at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:833)
	at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:387)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:419)
	at org.eclipse.jdt.internal.compiler.batch.Main.performCompilation(Main.java:4597)
	at org.eclipse.jdt.internal.compiler.batch.Main.compile(Main.java:1767)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.compile(JDTBasedSpoonCompiler.java:112)
	at spoon.TestBuildSpoonWithSpoon.compile(TestBuildSpoonWithSpoon.java:32)
	at spoon.TestBuildSpoonWithSpoon.buildSpoon(TestBuildSpoonWithSpoon.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.ClassNotFoundException: spoon.generating.meta.$Role$
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.findClass(CtTypeReferenceImpl.java:136)
	... 43 more
```

and so, the output which should contains `.class` of the test suite is empty.

Please find on my own fork, a branch that reproduce the bug: [danglotb/spoon](https://github.com/danglotb/spoon/tree/build-test-spoon).

Find the test [here](https://github.com/danglotb/spoon/blob/build-test-spoon/src/test/java/spoon/TestBuildSpoonWithSpoon.java#L17).

There is one test case that reproduce the errors, but you will have to fill the `String dependencies` with the output of `mvn dependency:build-classpath`, at the root folder of Spoon.

Cheers.

**Comment by monperrus** [comment]
@danglotb should we close this one?

**Comment by surli** [comment]
I got another error but not related with "$" symbol anymore: this has certainly been fixed in #1891
